docs/xss.rst

```diff
@@ -0,0 +1,95 @@
+Performing XSS Vulnerability Audits
+==================================
+
+The bok-choy framework includes the ability to perform XSS (cross-site scripting) audits on
+web pages using a short XSS locator defined in
+https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#XSS_Locator.
+
+Leverage Your Existing Tests and Fail on XSS Vulnerability Errors
+-------------------------------------------------------------
+
+To trigger XSS vulnerability audits passively, set the ``VERIFY_XSS``
+environment variable to ``True``. Doing so triggers an XSS audit
+whenever a page object's ``q`` method is called. If errors are
+found on the page, an XSSExposureError is raised.
+
+.. note:: XSS vulnerabilities will only be found if you have injected the specific string
+XSS_INJECTION defined in bok_choy.page_object into your page content.
+
+You might already have some bok-choy tests written for your web application.
+Here is an example of a bok-choy test that will implicitly check for XSS vulnerabilities.
+It involves clicking on a button on page, and it is assumed that the user's name is
+displayed on the page with the button. If the user name is not properly escaped, the display
+of the name (which is data provided by the user and thus potentially malicious) can cause
+XSS issues.
+
+.. code-block:: python
+
+    from bok_choy.page_object import PageObject, XSS_INJECTION
+
+
+    class MyPage(PageObject):
+        def url(self):
+            return 'https://www.mysite.com/page'
+
+        def click_button(self):
+            """
+            Click on the button element (id="button").
+            On my example page this will trigger an ajax call
+            that updates the #output div with the text "yes!"
+            """
+            self.q(css='div#fixture button').first.click()
+            self.wait_for_ajax()
+
+        @property
+        def output(self):
+            """
+            Return the contents of the "#output" div on the page.
+            """
+            text_list = self.q(css='#output').text
+
+            if len(text_list) < 1:
+                return None
+            else:
+                return text_list[0]
+
+    class MyPageTest(WebAppTest):
+        def setUp(self):
+            """
+            Log in as a particular user.
+            """
+            super(MyPageTest, self).setUp()
+            self.user_name = XSS_INJECTION
+            self.username, self.user_id = self.log_in_as_user(self.user_name)
+
+        def test_button_click_output(self):
+            page = MyPage(self.browser)
+            page.visit()
+            page.click_button()
+
+            self.assertEqual(page.output, 'yes!')
+
+        def log_in_user(self, user_name):
+            """
+            Would be implemented to log in as a particular user
+            with a potentially malicious, user-provided name.
+            """
+            pass
+
+Before running your bok-choy tests, set the environment variable
```

Comment:
I'm not sure if this was meant to be in the code example, or a comment after it? I don't think you need it because Step 2 above is where you're telling people to set the env var. You can include the code example with that step.